examples/js/loaders/GLTFLoader.js

```diff
@@ -1251,9 +1251,18 @@ THREE.GLTFLoader = ( function () {
 		// .extras has user-defined data, so check that .extras.targetNames is an array.
 		if ( meshDef.extras && Array.isArray( meshDef.extras.targetNames ) ) {
 
-			for ( var i = 0, il = meshDef.extras.targetNames.length; i < il; i ++ ) {
+			var targetNames = meshDef.extras.targetNames;
+			var dictionary = mesh.morphTargetDictionary;
 
-				mesh.morphTargetDictionary[ meshDef.extras.targetNames[ i ] ] = i;
+			for ( var key in dictionary ) {
+
+				if ( dictionary[ key ] < targetNames.length ) delete dictionary[ key ];
+
+			}
+
+			for ( var i = 0, il = targetNames.length; i < il; i ++ ) {
+
+				dictionary[ targetNames[ i ] ] = i;
 
 			}
```

Comment:
Maybe just reset `mesh.morphTargetDictionary = {}`? And because `extras.targetNames` is already an unofficial property, I think we should be very strict about what it can contain. If the length does not match, ignore the whole thing or even log an error.

```js
if ( Object.keys( dictionary ).length === targetNames.length ) {

  // ...

} else {

  console.warn( 'THREE.GLTFLoader: Invalid extras.targetNames length. Ignoring names.' );

}
```

Reply:
I like that idea because it's easier. Updated. Thanks.